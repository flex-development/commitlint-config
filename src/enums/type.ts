/**
 * @file Enums - Type
 * @module commitlint-config/enums/Type
 */

/**
 * Commit types.
 *
 * @see https://commitlint.js.org/#/reference-rules?id=type-enum
 *
 * @enum {Lowercase<string>}
 */
enum Type {
  BUILD = 'build',
  CHORE = 'chore',
  CI = 'ci',
  DOCS = 'docs',
  FEAT = 'feat',
  FIX = 'fix',
  PERF = 'perf',
  REFACTOR = 'refactor',
  REVERT = 'revert',
  STYLE = 'style',
  TEST = 'test',
  WIP = 'wip'
}

export default Type
