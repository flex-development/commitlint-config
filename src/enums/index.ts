/**
 * @file Entry Point - Enums
 * @module commitlint-config/enums
 */

export { default as PromptKind } from './kind-prompt'
export { default as ReferenceAction } from './reference-action'
export { default as Scope } from './scope'
export { default as Type } from './type'
