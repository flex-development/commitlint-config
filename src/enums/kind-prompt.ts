/**
 * @file Enums - PromptKind
 * @module commitlint-config/enums/PromptKind
 */

import type { PromptName } from '@commitlint/types'

/**
 * Prompt names.
 *
 * @see https://commitlint.js.org/#/reference-prompt
 *
 * @enum {PromptName}
 */
enum PromptKind {
  BODY = 'body',
  BREAKING = 'breaking',
  BREAKING_BODY = 'breakingBody',
  ISSUES = 'issues',
  ISSUES_BODY = 'issuesBody',
  IS_BREAKING = 'isBreaking',
  IS_ISSUE_AFFECTED = 'isIssueAffected',
  SCOPE = 'scope',
  SUBJECT = 'subject',
  TYPE = 'type'
}

export default PromptKind
