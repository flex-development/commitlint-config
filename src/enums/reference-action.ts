/**
 * @file Enums - ReferenceAction
 * @module commitlint-config/enums/ReferenceAction
 */

/**
 * Issue reference action types.
 *
 * @enum {Lowercase<string>}
 */
enum ReferenceAction {
  CLOSE = 'closes',
  FIX = 'fixes',
  RELEASE = 'releases',
  RESOLVE = 'resolves'
}

export default ReferenceAction
