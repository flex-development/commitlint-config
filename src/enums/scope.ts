/**
 * @file Enums - Scope
 * @module commitlint-config/enums/Scope
 */

/**
 * Commit scopes.
 *
 * @see https://commitlint.js.org/#/reference-rules?id=scope-enum
 *
 * @enum {Lowercase<string>}
 */
enum Scope {
  BUILD = 'build',
  CJS = 'cjs',
  CLI = 'cli',
  COMPONENTS = 'components',
  CONFIG = 'config',
  CONTROLLERS = 'controllers',
  DECORATORS = 'decorators',
  DEPS = 'deps',
  DEPS_BUNDLE = 'deps-bundle',
  DEPS_DEV = 'deps-dev',
  DEPS_OPT = 'deps-opt',
  DEPS_PEER = 'deps-peer',
  DTOS = 'dtos',
  ENTITIES = 'entities',
  ENUMS = 'enums',
  ESM = 'esm',
  EXCEPTIONS = 'exceptions',
  EXPORTS = 'exports',
  FACTORIES = 'factories',
  FILTERS = 'filters',
  GITHUB = 'github',
  GUARDS = 'guards',
  HOOKS = 'hooks',
  HYBRID = 'hybrid',
  INSTALL = 'install',
  INTERCEPTORS = 'interceptors',
  INTERFACES = 'interfaces',
  INTERNAL = 'internal',
  LIB = 'lib',
  LOADER = 'loader',
  MIDDLEWARE = 'middleware',
  MIXINS = 'mixins',
  MODELS = 'models',
  MODULES = 'modules',
  NAMESPACES = 'namespaces',
  NODE = 'node',
  NVM = 'nvm',
  PATCHES = 'patches',
  PIPES = 'pipes',
  PKG = 'pkg',
  PLUGINS = 'plugins',
  PONYFILL = 'ponyfill',
  PROPS = 'props',
  PROVIDERS = 'providers',
  PUBLIC = 'public',
  REGEX = 'regex',
  RELEASE = 'release',
  RULES = 'rules',
  SCRIPTS = 'scripts',
  SERVICES = 'services',
  SPELLING = 'spelling',
  STORYBOOK = 'storybook',
  STYLES = 'styles',
  SUBDOMAINS = 'subdomains',
  TESTS = 'tests',
  TS = 'ts',
  TS_NEXT = 'ts-next',
  TYPES = 'types',
  UTILS = 'utils',
  VERCEL = 'vercel',
  VITE = 'vite',
  VSCODE = 'vscode',
  WORKFLOWS = 'workflows',
  YARN = 'yarn'
}

export default Scope
