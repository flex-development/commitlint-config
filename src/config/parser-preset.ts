/**
 * @file Config - parserPreset
 * @module commitlint-config/config/parserPreset
 */

import { ReferenceAction } from '#src/enums'
import type { ParserPreset } from '#src/types'
import consola from 'consola'

/**
 * Commit parser preset.
 *
 * Pattern regex:
 *
 * - `headerPattern`: https://regex101.com/r/WAJeLp
 * - `revertPattern`: https://regex101.com/r/78TbQq
 *
 * @const {ParserPreset} parserPreset
 */
const parserPreset: ParserPreset = {
  name: 'conventional-changelog-conventionalcommits',
  parserOpts: {
    commentChar: '#',
    fieldPattern: /^-(?<field>.*?)-(?=\n*)$/m,
    headerCorrespondence: ['type', 'scope', 'breaking', 'subject', 'pr'],
    headerPattern:
      /^(?<type>[a-z]+)(?:\((?<scope>[a-z-]+)\))?(?<breaking>!)?: +(?<subject>(?:.+ \(#(?<pr>\d+)\))|.+)/i,
    issuePrefixes: ['#', 'GH-'],
    issuePrefixesCaseSensitive: true,
    mergeCorrespondence: null,
    mergePattern: null,
    noteKeywords: ['BREAKING-CHANGE'],
    referenceActions: Object.values(ReferenceAction),
    revertCorrespondence: [
      'type',
      'scope',
      'breaking',
      'owner',
      'repository',
      'user',
      'hash'
    ],
    revertPattern:
      /^(?<type>revert)(?:\((?<scope>[a-z-]+)\))?(?<breaking>!)?: (?:|(?:.+\n\n(?:- )?))(?:(?:(?:(?<owner>[\d*a-z~-][\w*.~-]*)\/(?<repository>[\da-z~-][\w.~-]*))|(?<user>[\da-z~-][\w.~-]*))@)?(?<hash>[\da-z]+)/,
    warn: consola.warn.bind(consola)
  }
}

export default parserPreset
