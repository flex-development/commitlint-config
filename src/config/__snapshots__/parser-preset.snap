// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:config/parserPreset > .parserOpts > .fieldPattern > should match commit field 1`] = `
{
  "0": "-gitTags-",
  "1": "gitTags",
  "groups": {
    "field": "gitTags",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .fieldPattern > should match commit field that ends with new line 1`] = `
{
  "0": "-hash-",
  "1": "hash",
  "groups": {
    "field": "hash",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .headerPattern > should match header with [type,breaking,subject,pr] 1`] = `
{
  "0": "refactor!: statements (#3)",
  "1": "refactor",
  "2": undefined,
  "3": "!",
  "4": "statements (#3)",
  "5": "3",
  "groups": {
    "breaking": "!",
    "pr": "3",
    "scope": undefined,
    "subject": "statements (#3)",
    "type": "refactor",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .headerPattern > should match header with [type,breaking,subject] 1`] = `
{
  "0": "refactor!: statements",
  "1": "refactor",
  "2": undefined,
  "3": "!",
  "4": "statements",
  "5": undefined,
  "groups": {
    "breaking": "!",
    "pr": undefined,
    "scope": undefined,
    "subject": "statements",
    "type": "refactor",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .headerPattern > should match header with [type,scope,breaking,subject,pr] 1`] = `
{
  "0": "refactor(utils)!: [getFormat] enforce absolute module id (#3)",
  "1": "refactor",
  "2": "utils",
  "3": "!",
  "4": "[getFormat] enforce absolute module id (#3)",
  "5": "3",
  "groups": {
    "breaking": "!",
    "pr": "3",
    "scope": "utils",
    "subject": "[getFormat] enforce absolute module id (#3)",
    "type": "refactor",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .headerPattern > should match header with [type,scope,breaking,subject] 1`] = `
{
  "0": "refactor(utils)!: [getSource] enforce absolute module id",
  "1": "refactor",
  "2": "utils",
  "3": "!",
  "4": "[getSource] enforce absolute module id",
  "5": undefined,
  "groups": {
    "breaking": "!",
    "pr": undefined,
    "scope": "utils",
    "subject": "[getSource] enforce absolute module id",
    "type": "refactor",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .headerPattern > should match header with [type,scope,subject,pr] 1`] = `
{
  "0": "build(deps-dev): Bump cspell from 6.25.0 to 6.26.1 (#2)",
  "1": "build",
  "2": "deps-dev",
  "3": undefined,
  "4": "Bump cspell from 6.25.0 to 6.26.1 (#2)",
  "5": "2",
  "groups": {
    "breaking": undefined,
    "pr": "2",
    "scope": "deps-dev",
    "subject": "Bump cspell from 6.25.0 to 6.26.1 (#2)",
    "type": "build",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .headerPattern > should match header with [type,scope,subject] 1`] = `
{
  "0": "build(deps-dev): bump @flex-development/mkbuild from 1.0.0-alpha.11 to 1.0.0-alpha.13",
  "1": "build",
  "2": "deps-dev",
  "3": undefined,
  "4": "bump @flex-development/mkbuild from 1.0.0-alpha.11 to 1.0.0-alpha.13",
  "5": undefined,
  "groups": {
    "breaking": undefined,
    "pr": undefined,
    "scope": "deps-dev",
    "subject": "bump @flex-development/mkbuild from 1.0.0-alpha.11 to 1.0.0-alpha.13",
    "type": "build",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .headerPattern > should match header with [type,subject,pr] 1`] = `
{
  "0": "release: 1.0.0 (#3)",
  "1": "release",
  "2": undefined,
  "3": undefined,
  "4": "1.0.0 (#3)",
  "5": "3",
  "groups": {
    "breaking": undefined,
    "pr": "3",
    "scope": undefined,
    "subject": "1.0.0 (#3)",
    "type": "release",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .headerPattern > should match header with [type,subject] 1`] = `
{
  "0": "docs: when should i use this?",
  "1": "docs",
  "2": undefined,
  "3": undefined,
  "4": "when should i use this?",
  "5": undefined,
  "groups": {
    "breaking": undefined,
    "pr": undefined,
    "scope": undefined,
    "subject": "when should i use this?",
    "type": "docs",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .revertPattern > should match header with [type,breaking,repo,hash] 1`] = `
{
  "0": "revert!: flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "1": "revert",
  "2": undefined,
  "3": "!",
  "4": "flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "5": "flex-development",
  "6": "ext-regex",
  "7": undefined,
  "8": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "groups": {
    "breaking": "!",
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "owner": "flex-development",
    "repository": "ext-regex",
    "scope": undefined,
    "subject": "flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "type": "revert",
    "user": undefined,
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .revertPattern > should match header with [type,breaking,user,hash] 1`] = `
{
  "0": "revert!: flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "1": "revert",
  "2": undefined,
  "3": "!",
  "4": "flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "5": undefined,
  "6": undefined,
  "7": "flexdevelopment",
  "8": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "groups": {
    "breaking": "!",
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "owner": undefined,
    "repository": undefined,
    "scope": undefined,
    "subject": "flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "type": "revert",
    "user": "flexdevelopment",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .revertPattern > should match header with [type,hash] 1`] = `
{
  "0": "revert: 4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "1": "revert",
  "2": undefined,
  "3": undefined,
  "4": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "5": undefined,
  "6": undefined,
  "7": undefined,
  "8": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "groups": {
    "breaking": undefined,
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "owner": undefined,
    "repository": undefined,
    "scope": undefined,
    "subject": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "type": "revert",
    "user": undefined,
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .revertPattern > should match header with [type,repo,hash] 1`] = `
{
  "0": "revert: flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "1": "revert",
  "2": undefined,
  "3": undefined,
  "4": "flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "5": "flex-development",
  "6": "ext-regex",
  "7": undefined,
  "8": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "groups": {
    "breaking": undefined,
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "owner": "flex-development",
    "repository": "ext-regex",
    "scope": undefined,
    "subject": "flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "type": "revert",
    "user": undefined,
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .revertPattern > should match header with [type,scope,breaking,repo,hash] 1`] = `
{
  "0": "revert(release)!: flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "1": "revert",
  "2": "release",
  "3": "!",
  "4": "flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "5": "flex-development",
  "6": "ext-regex",
  "7": undefined,
  "8": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "groups": {
    "breaking": "!",
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "owner": "flex-development",
    "repository": "ext-regex",
    "scope": "release",
    "subject": "flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "type": "revert",
    "user": undefined,
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .revertPattern > should match header with [type,scope,breaking,user,hash] 1`] = `
{
  "0": "revert(release)!: flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "1": "revert",
  "2": "release",
  "3": "!",
  "4": "flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "5": undefined,
  "6": undefined,
  "7": "flexdevelopment",
  "8": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "groups": {
    "breaking": "!",
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "owner": undefined,
    "repository": undefined,
    "scope": "release",
    "subject": "flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "type": "revert",
    "user": "flexdevelopment",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .revertPattern > should match header with [type,scope,hash] 1`] = `
{
  "0": "revert(release): 4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "1": "revert",
  "2": "release",
  "3": undefined,
  "4": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "5": undefined,
  "6": undefined,
  "7": undefined,
  "8": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "groups": {
    "breaking": undefined,
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "owner": undefined,
    "repository": undefined,
    "scope": "release",
    "subject": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "type": "revert",
    "user": undefined,
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .revertPattern > should match header with [type,scope,repo,hash] 1`] = `
{
  "0": "revert(release): flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "1": "revert",
  "2": "release",
  "3": undefined,
  "4": "flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "5": "flex-development",
  "6": "ext-regex",
  "7": undefined,
  "8": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "groups": {
    "breaking": undefined,
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "owner": "flex-development",
    "repository": "ext-regex",
    "scope": "release",
    "subject": "flex-development/ext-regex@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "type": "revert",
    "user": undefined,
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .revertPattern > should match header with [type,scope,user,hash] 1`] = `
{
  "0": "revert(release): flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "1": "revert",
  "2": "release",
  "3": undefined,
  "4": "flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "5": undefined,
  "6": undefined,
  "7": "flexdevelopment",
  "8": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "groups": {
    "breaking": undefined,
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "owner": undefined,
    "repository": undefined,
    "scope": "release",
    "subject": "flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "type": "revert",
    "user": "flexdevelopment",
  },
}
`;

exports[`unit:config/parserPreset > .parserOpts > .revertPattern > should match header with [type,user,hash] 1`] = `
{
  "0": "revert: flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "1": "revert",
  "2": undefined,
  "3": undefined,
  "4": "flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "5": undefined,
  "6": undefined,
  "7": "flexdevelopment",
  "8": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
  "groups": {
    "breaking": undefined,
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "owner": undefined,
    "repository": undefined,
    "scope": undefined,
    "subject": "flexdevelopment@4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "type": "revert",
    "user": "flexdevelopment",
  },
}
`;
